src/ring_theory/ideal/local_ring.lean

```diff
@@ -371,6 +371,51 @@ lemma map_comp (f : T →+* R) (g : R →+* S) [is_local_ring_hom f] [is_local_r
   (local_ring.residue_field.map g).comp (local_ring.residue_field.map f) :=
 ideal.quotient.ring_hom_ext $ ring_hom.ext $ λx, rfl
 
+section
+
+open ring_equiv local_ring
+
+@[simp] lemma map_id_apply {x : residue_field R} : map (ring_hom.id R) x = x :=
+by simp only [map_id, ring_hom.id_apply]
+
+@[simp] lemma map_map (f : R →+* S) (g : S →+* T) (x : residue_field R)
+[is_local_ring_hom f] [is_local_ring_hom g] :
+  map g (map f x) = map (g.comp f) x :=
+by simp only [map_comp, ring_hom.comp_apply]
+
+/-- A ring isomorphism defines an isomorphism of residue fields -/
+@[simps apply]
+noncomputable def map_equiv (f : R ≃+* S):
+  local_ring.residue_field R ≃+* local_ring.residue_field S :=
+{ to_fun := map (f : R →+* S),
+  inv_fun := map (f.symm : S →+* R),
+  left_inv := λ x, by simp only [map_map, symm_comp, map_id, ring_hom.id_apply],
+  right_inv := λ x, by simp only [map_map, comp_symm, map_id, ring_hom.id_apply],
+  map_mul' := ring_hom.map_mul _,
+  map_add' := ring_hom.map_add _ }
+
+@[simp]
+lemma map_equiv.symm (f : R ≃+* S):
+  (map_equiv f).symm = map_equiv f.symm := ext $ λ x, rfl
+
+/-- The group homomorphism from `ring_aut R` to `ring_aut k` where `k`
+is the residue field of `R`. -/
+@[simps]
+noncomputable def map_aut:
+  ring_aut R →* ring_aut (local_ring.residue_field R):=
+{ to_fun := map_equiv,
+  map_mul' := begin
```

Comment:
Can you extract this to a lemma about `map_equiv (e.trans f)`?

Reply:
OK, how should these new lemmata be named?

Reply:
I would guess `map_equiv_trans` and `map_equiv_refl` to match the `map_equiv_symm` you already have. Note that `map_equiv_trans` will be about `ring_equiv`s between three different rings, not about `ring_aut R`.